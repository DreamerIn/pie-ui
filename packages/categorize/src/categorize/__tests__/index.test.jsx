import { shallow } from 'enzyme';
import React from 'react';
import { Categorize } from '../index';

jest.mock('../id-context', () => ({
  withUid: jest.fn(a => a),
  Provider: jest.fn(a => a),
  generateId: jest.fn().mockReturnValue('1')
}));

describe('categorize', () => {
  let onAnswersChange;

  beforeEach(() => {
    onAnswersChange = jest.fn();
  });
  const wrapper = extras => {
    const defaults = {
      classes: {},
      session: {
        answers: []
      },
      model: {
        config: {
          choices: {
            position: 'top',
            columns: 2
          },
          categories: {
            columns: 2
          }
        },
        choices: [],
        categories: []
      },
      onAnswersChange
    };
    const props = { ...defaults, ...extras };
    return shallow(<Categorize {...props} />);
  };

  describe('snapshots', () => {
    it('renders', () => {
      expect(wrapper()).toMatchSnapshot();
    });

    it('incorrect', () => {
      expect(wrapper({ incorrect: true })).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    describe('dropChoice', () => {
      it('calls onAnswersChange', () => {
        const w = wrapper();

        w.instance().dropChoice('1', { id: '1', choiceIndex: 0 });
        expect(onAnswersChange).toBeCalledWith([
          { category: '1', choices: ['1'] }
        ]);
      });
    });

    describe('removeChoice', () => {
      it('calls onAnswersChange', () => {
        const w = wrapper({
          session: { answers: [{ category: '1', choices: ['1'] }] }
        });

        w.instance().removeChoice({ id: '1', categoryId: '1', choiceIndex: 0 });

        expect(onAnswersChange).toBeCalledWith([
          { category: '1', choices: [] }
        ]);
      });
    });
  });
});
