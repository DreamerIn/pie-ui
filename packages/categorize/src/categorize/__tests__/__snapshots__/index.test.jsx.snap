// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categorize snapshots incorrect 1`] = `
<mockConstructor
  value="1"
>
  <div>
    <_default
      onToggle={[Function]}
      show={false}
      toggled={false}
    />
    <div
      style={
        Object {
          "flexDirection": "column-reverse",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        grid={
          Object {
            "columns": NaN,
            "rows": NaN,
          }
        }
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
      <WithStyles(Choices)
        choicePosition="above"
        choices={Array []}
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
          }
        }
      />
    </div>
  </div>
</mockConstructor>
`;

exports[`categorize snapshots renders 1`] = `
<mockConstructor
  value="1"
>
  <div>
    <_default
      onToggle={[Function]}
      show={false}
      toggled={false}
    />
    <div
      style={
        Object {
          "flexDirection": "column-reverse",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        grid={
          Object {
            "columns": NaN,
            "rows": NaN,
          }
        }
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
      <WithStyles(Choices)
        choicePosition="above"
        choices={Array []}
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
          }
        }
      />
    </div>
  </div>
</mockConstructor>
`;

exports[`categorize snapshots renders with feedback 1`] = `
<mockConstructor
  value="1"
>
  <div>
    <_default
      onToggle={[Function]}
      show={false}
      toggled={false}
    />
    <div
      style={
        Object {
          "flexDirection": "column-reverse",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        grid={
          Object {
            "columns": NaN,
            "rows": NaN,
          }
        }
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
            "correctness": "correct",
            "feedback": Object {
              "correct": Object {
                "default": "Correct",
                "type": "default",
              },
              "incorrect": Object {
                "default": "Incorrect",
                "type": "default",
              },
              "partial": Object {
                "default": "Nearly",
                "type": "default",
              },
            },
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
      <WithStyles(Choices)
        choicePosition="above"
        choices={Array []}
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
            "correctness": "correct",
            "feedback": Object {
              "correct": Object {
                "default": "Correct",
                "type": "default",
              },
              "incorrect": Object {
                "default": "Incorrect",
                "type": "default",
              },
              "partial": Object {
                "default": "Nearly",
                "type": "default",
              },
            },
          }
        }
      />
    </div>
    <Feedback
      correctness="correct"
      feedback={
        Object {
          "correct": Object {
            "default": "Correct",
            "type": "default",
          },
          "incorrect": Object {
            "default": "Incorrect",
            "type": "default",
          },
          "partial": Object {
            "default": "Nearly",
            "type": "default",
          },
        }
      }
    />
  </div>
</mockConstructor>
`;
